lib/validate.js

```diff
@@ -94,8 +94,8 @@ exports.usingChecksum = (deviceFileDescriptor, options = {}) => {
     imageSize: options.imageSize,
     progress: options.progress,
     chunkSize: options.chunkSize
-  }).then((deviceChecksum) => {
-    if (options.imageChecksum !== deviceChecksum) {
+  }).then((results) => {
+    if (options.imageChecksum !== results.crc32) {
```

Comment:
I guess to remain backwards compatible we could inspect the argument type? If it's a single value assume it's a crc32, else it's a tuple of `(checksum_type, checksum)` ?